### PR TITLE
LPS-146419 View mode of a translated asset does shows default language

### PR DESCRIPTION
**Pull request by YurenaC**
Motivation
=======
Web contents can have several translations that should be accesibles from the information panel available within the Content Dashboard
[LPS-146419](https://issues.liferay.com/browse/LPS-145419)

Proposed Solution
========
Currently, the option to visualize those translations is available, but all translations lead to the web content version in the default language.

This PR contains a fix that properly generates the localized URLs to access each translation available

Steps to Verify:
----------------
_Requirements:_

1. A default Display Page Template created, containing a mapped 'Heading' fragment
2. One Web Content with a translation (e.g. Spanish)

_Step-by-Step:_

1. Open product menu > Applications > Content > Content Dashboard
2. At Content, hover over the asset > Click on 'Info' icon
3. At sidebar, expand 'Details' > Click on 'View' icon of the respective translation
4. The "view" version of the page should be the one matching the selected language

Screenshots:
-----------------------
webContentTranslations

### CODE REVIEW

**Comment by YurenaC** [comment]
Hi @cgoncas,
I'd like you to take a look at this code. There's something that still does not properly works, but the implementation should be pretty close to this one.

I debugged until the `FinderCacheImpl` and there, the code is not going through the code block that search in the database as  `emptyResult.matches(args`) (Method: getResult(FinderPath finderPath, Object[] args), Line: 196 aprox) 

The current implementation is quite similar to what we already have in `LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl`, `getCanonicalURL(LayoutDisplayPageObjectProvider layoutDisplayPageObjectProvider,	Locale locale)` method. The only difference is that the layoutDisplayPageObjectProvider is calculated there and not passed by parameter.

Please let me know your thoughts :)

**Comment by cgoncas** [comment]
Hey @YurenaC !!

Could you describe me the cases that are failing?

Thanks!!

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
ci:test:tango

**Comment by YurenaC** [comment]
ci:test:stable

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
ci:test:stable

**Comment by YurenaC** [comment]
Hi @cgoncas,
Could you please take a look at this PR? If look good to you, please, forward straight away, thanks!!

File: modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/ViewJournalArticleContentDashboardItemAction.java [hunk 1]
```diff
@@ -89,49 +108,146 @@ public String getURL(Locale locale) {
...
+	private String _getLocalizedURL(
+		List<LayoutSEOLink> localizedLayoutSEOLinks, Locale locale) {
+
+		List<LayoutSEOLink> result = new ArrayList<>();
+
+		ListUtil.filter(
+			localizedLayoutSEOLinks, result,
+			seoLink -> {
+				String stringLocale = locale.toString();
+
```
Comment (cgoncas) [inline modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/ViewJournalArticleContentDashboardItemAction.java:161]: You can compare Locales directly :)

Comment (YurenaC) [reply modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/ViewJournalArticleContentDashboardItemAction.java:161]: Once we have the seoLink, the HrefLang is a String, not a Locale. To compare them I have to write all that process

Comment (cgoncas) [reply modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/ViewJournalArticleContentDashboardItemAction.java:164]: Take a look to Language class, I think probably there is a method there you can use

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:stable

**Comment by YurenaC** [comment]
liferay/ci:test:relevant

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
> Hey @YurenaC !!
> 
> I think the back_url is missing, also you have deleted it from the test case.
> 
> I think that we can change the API so we return all the localized URLs.
> 
> Also, I think this change needs to be done for Documents And Blogs, maybe we can do a common class to share it?
> 
> Thanks!!

Hi @cgoncas,

-The backURL is fixed with my latest changes (test as well)
- I agree that should be generalized and applied to all assets in the Content Dashboard. I will create a new story for that and for the API change.

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
ci:forward

**Pull request closed without merging**